skbase/testing/utils/deep_equals.py

```diff
@@ -70,14 +70,21 @@ def ret(is_equal, msg):
     # we now know all types are the same
     # so now we compare values
 
+    # flag variables for available soft dependencies
+    pandas_available = _check_soft_dependencies("pandas", severity="none")
+    numpy_available = _check_soft_dependencies("numpy", severity="none")
+
+    if numpy_available:
```

Comment:
Should we factor numpy logic into helper function like in pandas? Answer could be that the logic is simpler, so it is fine to include directly.

Reply:
well, I was considering this, but I don't know how we would deal with lines 105ff, i.e., where `!=` results in an iterable, and if `numpy` is available, you want to use `np.all` (due to efficiency), otherwise `python` `all`.

Reply:
Ah, yea, lets leave as is then. We know it works this way and its limited upside.